corehq/motech/generic_inbound/models.py

```diff
@@ -72,3 +83,57 @@ def to_json(self):
             "expression_id": self.expression_id,
             "message": self.message,
         }
+
+
+class RequestLog(models.Model):
+
+    class Status(models.TextChoices):
+        FILTERED = 'filtered', _('Filtered')
+        VALIDATION_FAILED = 'validation_failed', _('Validation Failed')
+        SUCCESS = 'success', _('Success')
+        ERROR = 'error', _('Error')
+        REVERTED = 'reverted', _('Reverted')
+
+    class RequestMethod(models.TextChoices):
+        POST = 'POST'
+        PUT = 'PUT'
+        PATCH = 'PATCH'
+
+    id = models.UUIDField(primary_key=True, default=uuid4)
+    domain = models.CharField(max_length=255, db_index=True)
+    api = models.ForeignKey(ConfigurableAPI, on_delete=models.CASCADE)
+    status = models.CharField(max_length=32, choices=Status.choices, db_index=True)
+
+    timestamp = models.DateTimeField(auto_now=True, db_index=True)
+    attempts = models.PositiveSmallIntegerField(default=1)
+    response_status = models.PositiveSmallIntegerField(db_index=True)
+    error_message = models.TextField()
+
+    username = models.CharField(max_length=128, db_index=True)
+    request_method = models.CharField(max_length=32, choices=RequestMethod.choices)
+    request_query = models.CharField(max_length=8192)
+    request_body = models.TextField()
+    request_headers = models.JSONField(default=dict)
+    request_user_agent = models.TextField()
+    request_ip = models.GenericIPAddressField(db_index=True)
+
+    def __repr__(self):
+        return f"RequestLog(domain='{self.domain}', api={self.api}, status='{self.status}')"
+
+
+class ProcessingAttempt(models.Model):
+    log = models.ForeignKey(RequestLog, on_delete=models.CASCADE)
+    timestamp = models.DateTimeField(auto_now=True, db_index=True)
+    is_retry = models.BooleanField(default=False)
+    response_status = models.PositiveSmallIntegerField(db_index=True)
+    response_body = models.TextField()
+    raw_response = models.JSONField(default=dict)
+
+    xform_id = models.UUIDField(db_index=True, null=True, blank=True)
+    case_ids = ArrayField(models.UUIDField(default=uuid4), null=True, blank=True)
+
+    class Meta:
+        indexes = [GinIndex(fields=['case_ids'])]
```

Comment:
Think I should just drop it?  It would of course be nice to filter the report by case ID ("which requests updated this case"), but you could also get there by form ID, and that's the more fundamental transaction unit.  Like, if you're auditing a specific case, you'd find the form that made the change you're interested in, then come find the API request that submitted that form.

Reply:
Yea, I think we can drop the index. But we should probably make sure that you can easily identify the API config from the form. Currently I don't think we're doing much in that regard.

Reply:
Good point - I think a sensible way to approach that would be to use the `device_id` field for that.  That way, the XMLNS still identifies this as a case API submission, and the device ID would further clarify that this came from a particular configurable API.  We'd still have the user agent available in the log for auditing.  That seems in-line with the latest guidance on `submit_case_blocks`.

Reply:
On nice addition would be the form name which is what shows up on in the UI if it's present.